ext/mbstring/mbstring.c

```diff
@@ -2982,84 +2982,81 @@ static unsigned int mb_estimate_encoding_demerits(uint32_t w)
 	return 0;
 }
 
-/* When doing 'strict' detection, any string which is invalid in the candidate encoding
- * is rejected. With non-strict detection, we just continue, but apply demerits for
- * each invalid byte sequence */
-static const mbfl_encoding* mb_guess_encoding(unsigned char *in, size_t in_len, const mbfl_encoding **elist, unsigned int elist_size, bool strict)
+struct candidate {
+	const mbfl_encoding *enc;
+	unsigned char *in;
+	size_t in_len;
+	uint64_t demerits; /* Wide bit size to prevent overflow */
+	unsigned int state;
+};
```

Comment:
I'm terrible at struct packing, but is this the most compact way to define the struct?